hugr-core/src/macros.rs

```diff
@@ -57,22 +57,22 @@ pub(crate) use impl_box_clone;
 macro_rules! type_row {
     () => {
         {
-            $crate::types::TypeRow::new()
+            $crate::types::TypeRow::<false>::new()
         }
     };
     ($($t:expr),+ $(,)?) => {
         {
             use $crate::types;
             static ROW: &[types::Type] = &[$($t),*];
-            let row: types::TypeRow = ROW.into();
+            let row: types::TypeRow<false> = ROW.into();
```

Comment:
Hmmmm, this `<false>` is default anyway, so probably doesn't do anything. I should try `<_>` though, maybe we can get type inference to figure this out...

Reply:
`<_>` is rejected by `rustc` ("error[E0747]: type provided when a constant was expected"). Removed the unnecessary `<false>` though.